src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -710,6 +712,27 @@ public string GetUrlFromSerializableViewModel(ConsoleLogsPageState serializable)
 
     public ConsoleLogsPageState ConvertViewModelToSerializable()
     {
-        return new ConsoleLogsPageState(PageViewModel.SelectedResource?.Name);
+        var selectedResourceName = PageViewModel.SelectedOption.Id is not null
+            ? PageViewModel.SelectedOption.Name
+            // _noSelection, which doesn't have a resource attached to it
+            : null;
+        return new ConsoleLogsPageState(selectedResourceName);
+    }
+
+    private bool TryGetResourceByName(string resourceName, [NotNullWhen(true)] out ResourceViewModel? resource)
+    {
+        if (_resourceByName.TryGetValue(resourceName, out resource))
+        {
+            return true;
+        }
+
+        var resourcesWithDisplayName = _resourceByName.Values.Where(r => string.Equals(resourceName, r.DisplayName, StringComparisons.ResourceName)).ToList();
+        if (resourcesWithDisplayName.Count == 1)
+        {
+            resource = resourcesWithDisplayName.Single();
+            return true;
+        }
+
+        return false;
     }
```

Comment:
Could you add this as a help method onto ResourceViewModel. That way other people doing the same thing can reuse this logic.